source/daq/frequency_mask_trigger.cc

```diff
@@ -422,34 +435,65 @@ namespace psyllid
                     try
                     {
                         t_array_size = t_freq_data->get_array_size();
+
                         if( a_ctx.f_first_packet_after_start )
                         {
-                            LDEBUG( plog, "Resizing mask to " << t_array_size << " bins" );
-                            if( t_mask_buffer.size() != t_array_size ) t_mask_buffer.resize( t_array_size, 0. );
+                            if( t_mask_buffer.size() != t_array_size )
+                            {
+                                LWARN( plog, "Mask was not the right size; Resizing mask to " << t_array_size << " bins and filling it with zeros" );
+                                t_mask_buffer.resize( t_array_size, 0. );
+                            }
                             a_ctx.f_first_packet_after_start = false;
                         }
+
                         t_trigger_flag->set_flag( false );
+                        t_trigger_flag->set_high_threshold( false );
+                        t_trigger_flag->set_id( t_freq_data->get_pkt_in_session() );
 
                         for( unsigned i_bin = 0; i_bin < t_array_size; ++i_bin )
                         {
                             t_real = t_freq_data->get_array()[ i_bin ][ 0 ];
                             t_imag = t_freq_data->get_array()[ i_bin ][ 1 ];
-                /*#ifndef NDEBUG
-                            if( i_bin < 5 )
+                            t_power_amp = t_real*t_real + t_imag*t_imag;
+
+                            t_trigger_flag->set_id( t_freq_data->get_pkt_in_session() );
+
+                            if( high_threshold_factor == 1 )
```

Comment:
For this conditional, we should have a boolean that says whether or not to use the single threshold.  We shouldn't rely on an equality test with a floating-point number.